### ISSUE TITLE
incomplete between parameter type annotation and default value, but no error is reported.

### ISSUE THREAD

**Issue by frokaikan** [microsoft/TypeScript#61627]
### Acknowledgement

- [x] I acknowledge that issues using this template may be closed without further explanation at the maintainer's discretion.

### Comment

code:
```ts
interface I { }
class J { }
function X(arg1: number, arg2?: number, arg3: J = new J(), arg4?: I): number | null { return null; }
(arg1: number, arg2?: number, arg3: J = 11.33, arg4?: I): number | null => { return null; }
```
the `arg3` in the lambda expression has type J, but its default value is a number. However, typescript engine gives no error.

The version of `tsc` is 5.8.3

**Comment by MartinJohns** [comment]
Please use the [bug report](https://github.com/microsoft/TypeScript/blob/main/.github/ISSUE_TEMPLATE/bug_report.yml?plain=1) issue template for bug reports, or the [feature request](https://github.com/microsoft/TypeScript/blob/main/.github/ISSUE_TEMPLATE/feature_request.yml?plain=1) issue template for feature requests.

---

This is working as intended and mentioned in the FAQ:

- [Why are all types assignable to empty interfaces?
](https://github.com/microsoft/TypeScript/wiki/FAQ#why-are-all-types-assignable-to-empty-interfaces)
- [Why do these empty classes behave strangely?](https://github.com/microsoft/TypeScript/wiki/FAQ#why-do-these-empty-classes-behave-strangely)
